x-pack/plugins/ingest_manager/server/services/agents/actions.ts

```diff
@@ -15,12 +22,37 @@ export async function createAgentAction(
   soClient: SavedObjectsClientContract,
   newAgentAction: Omit<AgentAction, 'id'>
 ): Promise<AgentAction> {
-  const so = await soClient.create<AgentActionSOAttributes>(AGENT_ACTION_SAVED_OBJECT_TYPE, {
+  return createAction(soClient, newAgentAction);
```

Comment:
is it possible to overload createAgentAction or have it accept different types instead of having to create the new functions `createAgentPolicyAction` and `createAction`?